core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -307,6 +307,13 @@ private void beginSnapshot(ClusterState clusterState, final SnapshotsInProgress.
                 metaData = builder.build();
             }
 
+            // remove allocation ids from metadata
+            MetaData.Builder builder = MetaData.builder(metaData);
+            for (ObjectObjectCursor<String, IndexMetaData> index : metaData.getIndices()) {
+                builder.put(IndexMetaData.builder(index.value).removeAllPrimaryCandidateAllocationIds().build(), false);
```

Comment:
Upon restore we know the RestoreSource. In that case, we can have special handling. I remove this for now and keep this in mind when implementing the actual check whether allocation ids match.